ecs-cli/integ/cmd/compose.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2015-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ/stdout"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// A Project is the configuration needed to create an ECS Service.
+type Project struct {
+	Name       string
+	ConfigName string
+}
+
+// NewProject creates a new Project for an ECS Service.
+func NewProject(name string, configName string) *Project {
+	return &Project{
+		Name:       name,
+		ConfigName: configName,
+	}
+}
+
+// TestServiceUp runs `ecs-cli compose service up` for a project.
+func TestServiceUp(t *testing.T, p *Project) {
+	// Given
+	args := []string{
+		"compose",
+		"--project-name",
+		p.Name,
+		"service",
+		"up",
+		"--create-log-groups",
+		"--cluster-config",
+		p.ConfigName,
+	}
+	cmd := integ.GetCommand(args)
+
+	// When
+	out, err := cmd.Output()
+	if err != nil {
+		assert.FailNowf(t, "Failed to create service", "Error %v running %v", err, args)
+	}
+
+	// Then
+	stdout.Stdout(out).TestHasAllSnippets(t, []string{
```

Comment:
Hmm. I kind of think that it would be better to call ecs:DescribeService to verify that the command succeeded, rather than checking the ECS CLI output.

Reply:
I'm fine with just validating the command output, but I'd prefer if in addition we also made an API call to verify/check.

Reply:
Yeah, so I also debated about this as well. I wrote the tests from the point of view of a user. How do they validate if `compose service scale` worked? they probably run `compose service ps`. That's why `fargate_service_test` is structured like:
```go
// Increase the number of running tasks
cmd.TestServiceScale(t, project, 2)
cmd.TestServicePs(t, project, 2)
```
In scenarios where it didn't seem satisfying, like in `TestUp` I use the `cfnClient` to validate that the stack's name is following our naming template. 

I couldn't think of some integration to test behind the scenes to validate that it's doing the right thing that PS doesn't do for us. I'm definitely open to suggestions here though :)

Reply:
Okay, that makes sense to me. This satisfies my concern :)